tests/execution_engine/test_sparkdf_execution_engine.py

```diff
@@ -732,16 +745,78 @@ def mocked_reader_function(*args, **kwargs):
     assert len(test_sparkdf.columns) == 2
 
 
-def test_get_batch_with_split_on_whole_table_runtime(
-    test_sparkdf, basic_spark_df_execution_engine
+def test_get_batch_with_split_on_whole_table_azure(
+    spark_session, basic_spark_df_execution_engine
 ):
-    test_sparkdf = basic_spark_df_execution_engine.get_batch_data(
-        RuntimeDataBatchSpec(
-            batch_data=test_sparkdf, splitter_method="_split_on_whole_table"
+    # noinspection PyUnusedLocal
+    def mocked_get_reader_function(*args, **kwargs):
+        # noinspection PyUnusedLocal,PyShadowingNames
+        def mocked_reader_function(*args, **kwargs):
+            pd_df = pd.DataFrame({"a": [1, 2, 3, 4], "b": [2, 3, 4, None]})
+            df = spark_session.createDataFrame(
+                [
+                    tuple(
+                        None if isinstance(x, (float, int)) and np.isnan(x) else x
+                        for x in record.tolist()
+                    )
+                    for record in pd_df.to_records(index=False)
+                ],
+                pd_df.columns.tolist(),
+            )
+            return df
+
+        return mocked_reader_function
+
+    spark_engine = basic_spark_df_execution_engine
+    spark_engine._get_reader_fn = mocked_get_reader_function
+
+    test_sparkdf = spark_engine.get_batch_data(
+        AzureBatchSpec(
+            path="wasbs://test_container@test_account.blob.core.windows.net/test_dir/test_file.csv",
+            reader_method="csv",
+            reader_options={"header": True},
+            splitter_method="_split_on_whole_table",
         )
     ).dataframe
-    assert test_sparkdf.count() == 120
-    assert len(test_sparkdf.columns) == 10
+    assert test_sparkdf.count() == 4
+    assert len(test_sparkdf.columns) == 2
```

Comment:
Should we be adding this in here given that this is a GCS PR?

Reply:
@cdkini I believe this was just me moving the test functions in order to organize them better in the module.  Thanks.